core/src/test/java/org/apache/iceberg/TestScansAndSchemaEvolution.java

```diff
@@ -70,7 +70,8 @@ public TestScansAndSchemaEvolution(int formatVersion) {
   private DataFile createDataFile(String partValue) throws IOException {
     List<GenericData.Record> expected = RandomAvroData.generate(SCHEMA, 100, 0L);
 
-    OutputFile dataFile = new InMemoryOutputFile(FileFormat.AVRO.addExtension(UUID.randomUUID().toString()));
+    OutputFile dataFile =
+        new InMemoryFileIO().newOutputFile(FileFormat.AVRO.addExtension(UUID.randomUUID().toString()));
```

Comment:
Should we store the `InMemoryFileIO` in its own variable so that the underlying `store` is only instantiated once?